ethereumkit/src/main/java/org/p2p/ethereumkit/external/repository/EthereumKitRepository.kt

```diff
@@ -147,25 +132,44 @@ internal class EthereumKitRepository(
 
     private suspend fun loadTokensMetadata(): List<EthTokenMetadata> {
         val publicKey = tokenKeyProvider?.publicKey ?: throwInitError()
-        val tokenAddresses = ERC20Tokens.values().map { it.contractAddress }
 
-        val tokensBalances = loadTokenBalances(publicKey, tokenAddresses.map(::EthAddress))
+        val erc20TokensAddresses = ERC20Tokens.values().map { it.contractAddress }
+
+        val allTokensAddresses = buildList<String> {
+            this += TOKEN_SERVICE_NATIVE_ETH_TOKEN
+            this += erc20TokensAddresses
+        }
+
+        val tokensBalances = loadTokenBalances(publicKey, erc20TokensAddresses.map(::EthAddress))
 
         val tokensMetadata = tokenServiceRepository.loadMetadataForTokens(
             chain = TokenServiceNetwork.ETHEREUM,
-            tokenAddresses = tokenAddresses
-        ).map { metadata ->
-            val ethAddress = EthAddress(metadata.address)
-            val tokenBalance = tokensBalances.firstOrNull { it.contractAddress == ethAddress }
+            tokenAddresses = allTokensAddresses
+        )
+        val ethTokenMetadata = tokensMetadata.firstOrNull {
+            it.address == TOKEN_SERVICE_NATIVE_ETH_TOKEN
+        } ?: return emptyList()
+        val ethBalance = getBalance()
+        val ethToken = converter.toEthTokenMetadata(
```

Comment:
nativeEthTokenMetadata

and building nativeEthTokenMetadata can be extracted to a method